fedgov-cv-ISDcreator-webapp/src/main/webapp/js/depositor.js

```diff
@@ -310,6 +312,15 @@ function createMessageJSON()
             "speedLimit": box.features[i].attributes.speedLimit,
             "drivingLanes": drivingLaneArray
         };
+        
+        if (approachArray[i].approachType === undefined) {
+            incompleteApproaches.push(lanes.features[i].attributes.laneNumber)
+
+            if (incompleteApproaches) {
+                $("#message_deposit").prop('disabled', true);
+                $('#alert_placeholder').html('<div id="approach-alert" class="alert alert-warning alert-dismissable"><button type="button" class="close" data-dismiss="alert" aria-hidden="true">&times;</button><span>'+ "Approach Type empty for approach associated with lane(s) " + incompleteApproaches.toString() + "." +'</span></div>');

```

Comment:
Since approach type is a mandatory field, I believe we can throw a required alert using the "alert-danger" class instead of alert warning.

Further, we can also grey out the Encode button so that the approach type is provided before running into Null Pointer exception.